stackrox/stackrox
# ROX-19298: Replace violationTypes with frontend alert proto types

## Description

* Follow up task: left-over from long ago recently noticed in #7411
* Postponed to avoid merge conflicts with changes for cherry-pick into 4.2.0 release branch in #7737
* Prerequisite to add conditional RBAC for violation tabs.

### Analysis

1. src/services/AlertsService.ts at global scope imports from violationsTypes.ts at container scope instead of src/types/alert.proto.ts file at global scope.
2. violationsTypes.ts imports and re-exports types from src/types/policy.proto.ts file at global scope.

### Solution

1. src/services/AlertsService.ts imports from files which correspond to and are kept up-to-date with backend response types:
    * src/types/alerts.proto.ts https://github.com/stackrox/stackrox/blob/master/proto/storage/alert.proto
    * src/types/policy.proto.ts https://github.com/stackrox/stackrox/blob/master/proto/storage/policy.proto
2. Violations components import directly from the types files.
3. Replace JavaScript-style tests whether optional properties are truthy with TypeScript-style tests whether optional property is `in` object to distinguish subtypes.

## Checklist

## Testing Performed

Delete redundant types does not decrease size. Replace JavaScript with TypeScript tests for optional properties does increase size somewhat.

1. `yarn lint` in ui
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.css: 0 = 3910667 - 3910667
        total: 40 = 10156980 - 10156940
    * `ls -al build/static/js/*.js | wc -l`
        0 = 83 - 83 files
3. `yarn start` in ui

### Manual testing

1. Visit /main/violations and see `DeploymentListAlert` response
    ![violations](https://github.com/stackrox/stackrox/assets/11862657/095565ba-358b-422b-801c-9f6eff88fe51)

2. Click a violation to visit its page and see `DeploymentAlert` response
    ![violation](https://github.com/stackrox/stackrox/assets/11862657/9cf074aa-8bd5-44b4-a594-d11fb5f660d2)

### Integration testing

* violations/violations.test.js
